draft-specifications/Model.md

```diff
@@ -0,0 +1,326 @@
+> order, though this is not a concern of the abstract syntax and data model.
+>
+> A JavaScript object that owns any non-string keys (numbers or symbols) or
+> with a prototype other than `Object.prototype cannot be passed as an OCapN
+> struct as all such objects are reserved for concrete representations of OCapN
+> types.
+> The key `Symbol.for('passStyle')` is special and indicates the kind of OCapN
+> value the object represents.
+
+## Tagged
+
+A value with a string tag.
+
+> OCapN provides a small number of container types with minimal semantics.
+> Tags allow protocols to emerge from values annotated to indicate a richer
+> variety of semantics.
+> Concretely, a list might be differentiated from or hinted to be an ordered
+> list, sorted list, or a set of unique values.
+>
+> - **Guile**: currently a Guile record labeled `'(syrec ...)` (**imported**)
+> - **JavaScript**: an object with the key `Symbol.for('passStyle')` and value
+>   `tagged`, with a string tagName and a single `"payload"` field carrying the tagged value.
+>   ```js
+>   {
+>     [Symbol.for('passStyle')]: 'tagged',
+>     [Symbol.toStringTag]: tagName,
+>     payload
+>   }
+>   ```
+> - **Python**: `Tagged(tag, value)` where `tag` is a `string` and `Tagged` is
+>   imported from `ocapn`.
+>
+> OCapN tagged data have a single value (are not variadic) [November 14, 2023](https://github.com/ocapn/ocapn/issues/94)
+>
+> What are tagged values for? https://github.com/ocapn/ocapn/issues/52
+
+## Remotable
+
+A value that can receive messages.
+
+Remotable values can be either a local value or a remote value that forwards
+messages to a local value.
+The latter is a presence: a remotable that forwards messages to the peer where the
+message can be delivered to its corresponding local remotable.
+
+Comparing the identities of remotable values is pass-invariant, meaning that
+comparing the identities of any pair of remotable values will produce the same
+result on one peer if passed over OCapN to another peer and compared there,
+regardless of whether either value is a presence on either peer.
+
+Although Remotables and Promises are both Capabilities for delivering messages,
+Promises do not support pass-invariant identity comparison.
+
+> - **Guile**: a function
+> - **JavaScript**: A *far object* or a *remote presence* of that far object.
+>   ```js
+>   {
+>     __proto__: {
+>       [Symbol.for('passStyle')]: 'remotable',
+>       [Symbol.toStringTag]: allegedInterfaceName,
+>     },
+>     ...methods
+>   }
+>   ```
+> - **Python**: to be proposed
+>
+> Tracking: https://github.com/ocapn/ocapn/issues/49
+
+## Promise
+
+A promise represents the eventual return value (fulfillment) or thrown error
+(rejection reason) for a message delivery.
+
+Messages delivered to a promise are forwarded to their eventual fulfillment
+value or rejected with the promises's eventual rejection reason.
+
+> - **Guile**: to be proposed
```

Comment:
There's no promise API in Guile. It's a Goblins addition.

Reply:
Is this adequate?
```suggestion
> - **Guile**: a Goblins promise
```